docs/tutorial.rst

```diff
@@ -306,17 +318,303 @@ trees. In this example we incorrectly inferred that 0 coalesces with the
 ancestor of nodes 2, 3, 4 and 5 before 1.
 
 
-.. todo::
+************
+Data example
+************
+
+Inputting real data for inference is similar in principle to the examples above:
+you simply iterate over the sites in your data file, calling :meth:`.SampleData.add_site`
+for each variable site. We do not provide methods to read directly from other formats, as
+these formats are often too complex for automatic conversion. However, many Python
+libraries are available for reading different genomic data formats, and writing a script
+to iterate over the sites in such files is usually a fairly simple task. The example
+below shows how this can be done for a VCF file, using a freely available VCF reading
+library.
+
++++++++++++++
+Reading a VCF
++++++++++++++
+
+A common way to store genetic variation data is as a VCF (Variant Call Format) file.
+The `CYVCF2 library <https://github.com/brentp/cyvcf2>`_ provides a relatively fast
+Python interface for reading VCF files. The following script demonstrates how to infer a
+tree sequence from a simple VCF file. In particular the function below named
+``add_diploid_sites`` illustrates how to iterate over the variants in a ``CYVCF2.VCF``
+object and add them to a ``tsinfer`` sample data file, while printing out the progression
+through the file in number of base pairs. This particular script assumes that the VCF
+describes a set of phased, diploid individuals, and has the ancestral state present in
+the INFO field of the VCF (otherwise it takes the reference allele as the ancestral
+state). For example data, we use a publicly available VCF file of the genetic variants
+from chromosome 24 of ten Norwegian and French house sparrows, *Passer domesticus*
+(thanks to Mark Ravinet for the data file):
 
-    1. Add documentation links for msprime above so we can explain tree
-       sequences there.
+.. code-block:: python
 
-++++++++++++
-Data example
-++++++++++++
+    import cyvcf2
+    import tsinfer
+    import tqdm
+
+    def add_diploid_sites(vcf, samples):
+        """
+        Loop over sites in the vcf and add them to the samples object, reordering the
+        alleles to put the ancestral allele first, if it is available.
+        """
+        progressbar = tqdm.tqdm(total=samples.sequence_length, desc="Read VCF", unit='bp')
```

Comment:
I don't think we need the progressbar --- it's not necessary here for reading this VCF.

Reply:
OK. Although I think that the version with a progressbar should be described or demoed *somewhere*. Both Mark and Wenhua are testing their files using some derivation of this script, and will try to apply it to their own VCFs, when the progressbar will come in handy. I added it because I wanted to test on the same samples using chromosome 8, and I couldn't work out if the script had hung or not. So I think most people would benefit from some sort of feedback. 

We could, of course, just *state* in the text that you can add a progressbar by getting the sequence length out of the VCF and updating the value based on the genomic position. That way we would keep it out of the demo code...

Reply:
Or we could add a separate snippet showing this.

Reply:
Can't see an easy way to add a snippet. I suggest:

```
.. note::
    Larger VCF files will take some time to parse, and you may wish to add some sort of
    progress indicator to this script, for example by inserting the line
    `progressbar = tqdm.tqdm(total=samples.sequence_length, desc="Read VCF", unit='bp')`
    at the top of the `add_diploid_sites` function, followed by
    `progressbar.update(variant.POS - pos)` as the first line of the variant loop within
    that function.
```

Reply:
(p.s. I do note that you use a tqdm progressbar in the simulation example higher up in the file, so it's not unfamiliar territory in this tutorial)